server/rest/src/main/java/org/infinispan/rest/stream/CacheEntryStreamProcessor.java

```diff
@@ -0,0 +1,133 @@
+package org.infinispan.rest.stream;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.commons.dataconversion.internal.Json;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.reactivestreams.Publisher;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * A {@link CacheChunkedStream} that reads {@link Map.Entry} and produces a JSON output.
+ * For example:
+ * <p>
+ * <pre>{@code
+ * [
+ *   {"key":1,"value":"value","timeToLiveSeconds": -1, "maxIdleTimeSeconds": -1, "created": -1, "lastUsed": -1, "expireTime": -1},
+ *   {"key":2,"value":"value2","timeToLiveSeconds": -1, "maxIdleTimeSeconds": -1, "created": -1, "lastUsed": -1, "expireTime": -1}
+ * ]
+ * }</pre>
+ *
+ *
+ * @since 12.0
+ */
+public class CacheEntryStreamProcessor extends CacheChunkedStream<CacheEntry<?, ?>> {
+   private final boolean keysAreJson;
+   private final boolean valuesAreJson;
+
+   private static final byte[] KEY_LABEL = "\"key\":".getBytes();
+   private static final byte[] VALUE_LABEL = "\"value\":".getBytes();
+
+   private static final byte OPEN_ITEM_CHAR = '{';
+   private static final byte SEPARATOR = ',';
+   private static final byte CLOSE_ITEM_CHAR = '}';
+   private final boolean includeMetadata;
+
+   public CacheEntryStreamProcessor(Publisher<CacheEntry<?, ?>> publisher, boolean keysAreJson, boolean valuesAreJson,
+         boolean includeMetadata) {
+      super(publisher);
+      this.keysAreJson = keysAreJson;
+      this.valuesAreJson = valuesAreJson;
+      this.includeMetadata = includeMetadata;
+   }
+
+   @Override
+   public void subscribe(ChannelHandlerContext ctx) {
+      publisher.subscribe(new EntrySubscriber(ctx, ctx.alloc()));
+   }
+
+   class EntrySubscriber extends ByteBufSubscriber<CacheEntry<?, ?>> {
+
+      protected EntrySubscriber(ChannelHandlerContext ctx, ByteBufAllocator allocator) {
+         super(ctx, allocator);
+      }
+
+      @Override
+      void writeItem(CacheEntry<?, ?> item, ByteBuf pending) {
+         byte[] keyBytes = escape(item.getKey(), keysAreJson);
+         byte[] valueBytes = escape(item.getValue(), valuesAreJson);
+         byte[] metadataBytes = includeMetadata ? metadataBytes(item) : null;
+
+         // 1 for {
+         // Key label
+         // Key bytes
+         // 1 for ,
+         // Value label
+         // Value bytes
+         // 1 for }
+         int bytesRequired = KEY_LABEL.length + keyBytes.length + 1 + VALUE_LABEL.length + valueBytes.length + 1;
```

Comment:
Missing the 1 for `{`.

```suggestion
         int bytesRequired = 1 + KEY_LABEL.length + keyBytes.length + 1 + VALUE_LABEL.length + valueBytes.length + 1;
```

Reply:
Ah, yes. I forgot to add it back in when I put the comment, good catch.